pkg/cluster/components/version.go

```diff
@@ -0,0 +1,54 @@
+package components
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/blang/semver"
+)
+
+// Version represents versioned cluster information,
+// including the semver information and container
+// image information.
+type Version struct {
+	// Semver is the semver parsed version for comparisons.
+	Semver semver.Version
```

Comment:
nit: Having both `Semver` and `Image` be public means they could become out of sync. Maybe we just expose a `Semver()` method which extracts from Image.